dGame/EntityManager.cpp

```diff
@@ -161,111 +161,82 @@ void EntityManager::DestroyEntity(Entity* entity) {
 }
 
 void EntityManager::UpdateEntities(const float deltaTime) {
-	for (const auto& e : m_Entities) {
-	    e.second->Update(deltaTime);
+	for (auto entry = m_Entities.begin(); entry != m_Entities.end(); entry++) {
```

Comment:
Not sure this form of iteration is any more efficient than `for (var i : list)`. Regardless: `const auto& entry` instead of just `auto entry` avoids a copy here.

Reply:
can't be `const auto&` since we are modifying the value of the iterator to traverse the list

Reply:
It can be passed by reference here though with &

Reply:
Further discussion shows we cant actually pass by reference here at the least